controller-server/src/main/java/com/yahoo/vespa/hosted/controller/application/ApplicationPackage.java

```diff
@@ -49,22 +49,40 @@
     private final Optional<Version> compileVersion;
     private final Optional<Instant> buildTime;
     private final List<X509Certificate> trustedCertificates;
-    
-    /** 
+
+    /**
      * Creates an application package from its zipped content.
      * This <b>assigns ownership</b> of the given byte array to this class;
      * it must not be further changed by the caller.
      */
     public ApplicationPackage(byte[] zippedContent) {
+        this(zippedContent, false);
+    }
+
+    /**
+     * Creates an application package from its zipped content.
+     * This <b>assigns ownership</b> of the given byte array to this class;
+     * it must not be further changed by the caller.
```

Comment:
Consider copying the array, like we do in other classes.

Reply:
I guess this isn't done because the array can be pretty large?

Reply:
Yes.